scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py

```diff
@@ -139,17 +139,46 @@ def _merge_workspace_with_limits(focused_ws, q_lims):
     min_x = np.inf
     max_x = -np.inf
     num_x = -np.inf
-    for ws in focused_ws:
-        x_data = ws.dataX(0)
+    ws_max_range = 0
+    largest_range_spectrum = 0
+    for i in range(focused_ws.size()):
```

Comment:
Since we are now calculating these we should really hoist everything from line 139:152 into a separate function and unit test it all.

We can rely on Python duck typing to act in place of a workspace:
```
class MockedWs(object):
    def size():
        return 1

    def dataX(_):
        return [0, 1, 2]
```

Then we should test that we get the expected values for a sane ws, a workspace full or zeros and infs (which can sometimes happen if someone screws the reduction params)